### PR TITLE
Feedback implementation

### PR DESCRIPTION
**Pull request by Katerina-codes**
Hello! @MollieS @Maikon @sarahabimay
Here are the changes I have made to my Tic Tac Toe Game so far. There are more to come!

- Created a replay feature.
- Added public and private methods and updated tests to support this.
- Corrected one line of indentation and unnecessary variable assignment.
- Used each_with_index and reverse to get the diagonals dynamically.

Following the feedback, I also made the changes below but accidentally made them on master.

- Removed old grid format and replaced it with an array of numbers from 1 - 9.
- Refactored column_win method using transpose.

### CODE REVIEW

File: lib/computer_player.rb [hunk 1]
```diff
@@ -35,50 +37,48 @@ def play_move(move_sequences, player_mark, opponent_mark, current_grid)
...
   end
 
   def possible_fork?(current_grid, opponent_mark, value_indexes, number)
-    index_a = value_indexes[0][0]
-    index_a1 = value_indexes[0][1]
-    index_b = value_indexes[1][0]
-    index_b1 = value_indexes[1][1]
-    current_grid[index_a][index_a1].include?(opponent_mark) && current_grid[index_b][index_b1].include?(opponent_mark) && @grid.get_available_moves(current_grid).include?(number)
+    index_a = value_indexes[0]
+    index_a1 = value_indexes[1]
```
Comment (Maikon) [inline lib/computer_player.rb]: What does `index_a` and `index_a1` mean? What does `value_indexes` represent?

File: lib/computer_player.rb [hunk 2]
```diff
@@ -35,50 +37,48 @@ def play_move(move_sequences, player_mark, opponent_mark, current_grid)
...
+    current_grid[index_a] == opponent_mark && current_grid[index_a1] == opponent_mark && @grid.get_available_moves(current_grid).include?(number)
```
Comment (Maikon) [inline lib/computer_player.rb]: Try to wrap each condition in a method with a meaningful name that reveals intent. Will probably make reading this easier.

File: lib/computer_player.rb [hunk 3]
```diff
@@ -35,50 +37,48 @@ def play_move(move_sequences, player_mark, opponent_mark, current_grid)
...
       6
-    elsif possible_fork?(current_grid, opponent_mark, [[0, 2], [2, 0]], 4)
+    elsif possible_fork?(current_grid, opponent_mark, [2, 6], 4)
       4
     else
       no_fork
     end
   end
 
+  def play_five_as_first_move
```
Comment (Maikon) [inline lib/computer_player.rb]: `center_move`?

File: lib/converter.rb [hunk 4]
```diff
@@ -1,28 +1,7 @@
...
-      "| #{mark} "
-    elsif move == 2 || move == 5 || move == 8
-      "| #{mark} |"
-    else
-      " #{mark} |"
-    end
+    move -= 1
   end
 
 end
```
Comment (Maikon) [inline lib/converter.rb]: Is this class really needed for this line `move -= 1`? Could it live perhaps elsewhere more relevant?

File: lib/game.rb [hunk 5]
```diff
@@ -8,62 +8,21 @@ def initialize(input_output, grid, converter, human_player, computer_player)
...
 
-  def get_end_score(current_grid, p1_mark, p2_mark)
-    if @grid.does_any_row_have_three_marks?(current_grid, p1_mark) || @grid.does_any_column_have_three_marks?(current_grid, p1_mark) || @grid.does_either_diagonal_have_three_marks?(current_grid, p1_mark)
-      @input_output.display_player_one_wins
-    elsif @grid.does_any_row_have_three_marks?(current_grid, p2_mark) || @grid.does_any_column_have_three_marks?(current_grid, p2_mark) || @grid.does_either_diagonal_have_three_marks?(current_grid, p2_mark)
-      @input_output.display_player_two_wins
+  def replay
+    @input_output.ask_for_replay
+    replay_choice = @input_output.get_valid_replay_decision
+    if replay_choice == 1
```
Comment (Maikon) [inline lib/game.rb]: `1` is a magic number. What does `1` actually represent?

File: lib/game.rb [hunk 6]
```diff
@@ -77,9 +36,9 @@ def game_flow
     p2_mark = "O"
     p1_type, p2_type = player_types[game_mode]
     winning_move_sequences = [[1, 2, 3], [4, 5, 6], [7, 8, 9], [1, 4, 7], [2, 5, 8], [3, 6, 9], [1, 5, 9], [3, 5, 7]]
-    until game_is_over?(winning_move_sequences, p1_winning_move_sequence) || game_is_over?(winning_move_sequences, p2_winning_move_sequence) || @grid.get_available_moves(current_grid) == []
+    until game_over?(winning_move_sequences, p1_winning_move_sequence) || game_over?(winning_move_sequences, p2_winning_move_sequence) || @grid.get_available_moves(current_grid) == []
```
Comment (Maikon) [inline lib/game.rb]: Maybe leave this for another PR but consider why is it necessary to check if the game is over multiple times and what it would take to only have to check once.

Comment (Maikon) [inline lib/game.rb]: The `[]` is again missing context. What does it represent?

File: lib/game.rb [hunk 7]
```diff
@@ -77,9 +36,9 @@ def game_flow
...
       winning_move_sequences = player_move_flow(p1_type, p1_mark, p2_mark, winning_move_sequences, current_grid)
-      if game_is_over?(winning_move_sequences, p1_winning_move_sequence) || @grid.get_available_moves(current_grid) == []
+      if game_over?(winning_move_sequences, p1_winning_move_sequence) || @grid.get_available_moves(current_grid) == []
```
Comment (Maikon) [inline lib/game.rb]: Same comments as above line.

**Review by Maikon — COMMENTED** [review]
Reviewed some of it so far but good job. Some comments below (will add more when I review more of it)

File: lib/converter.rb (hunk 4 above)
Comment (Katerina-codes) [reply lib/converter.rb]: Agreed, was trying to remove this yesterday but things were breaking. I will investigate.

File: lib/computer_player.rb (hunk 2 above)
Comment (Katerina-codes) [reply lib/computer_player.rb]: Updated.

File: lib/computer_player.rb (hunk 3 above)
Comment (Katerina-codes) [reply lib/computer_player.rb]: Updated.

File: lib/game.rb (hunk 5 above)
Comment (Katerina-codes) [reply lib/game.rb]: Updated.

File: lib/game.rb (hunk 6 above)
Comment (Katerina-codes) [reply lib/game.rb]: Each player has different move_sequences which is why it's checked twice. However, you are right and this was one of the things I wanted remove. Once I remove move_sequences then I'll be able to use the grid throughout.

Comment (Katerina-codes) [reply lib/game.rb]: Updated.

File: lib/game.rb (hunk 7 above)
Comment (Katerina-codes) [reply lib/game.rb]: Updated.

File: lib/computer_player.rb (hunk 1 above)
Comment (Katerina-codes) [reply lib/computer_player.rb]: Updated.
